Add text for `readonly` member

BillWagner opened the pull request:
Fixes #928

The PR for readonly members in structs did not add text that a temporary copy of `this` wasn't required to invoke a readonly member. (The clause in structs did state that a copy wasn't required, but that left the spec in an inconsistent state.

BillWagner committed:
Add text for `readonly` member

Fixes #928

The PR for readonly members in structs did not add text that a temporary copy of `this` wasn't required to invoke a readonly member. (The clause in structs did state that a copy wasn't required, but that left the spec in an inconsistent state.

jskeet commented on standard/expressions.md:
```diff
@@ -1217,7 +1217,7 @@ The run-time processing of a function member invocation consists of the followin
 - Otherwise, if the type of `E` is a value-type `V`, and `M` is declared or overridden in `V`:
   - `E` is evaluated. If this evaluation causes an exception, then no further steps are executed. For an instance constructor, this evaluation consists of allocating storage (typically from an execution stack) for the new object. In this case `E` is classified as a variable.
-  - If `E` is not classified as a variable, or if `V` is not a readonly struct type ([§16.2.2](structs.md#1622-struct-modifiers)), and `E` is one of:
+  - If `E` is not classified as a variable, or if `V` is not a readonly struct type ([§16.2.2](structs.md#1622-struct-modifiers)) and the declaration of `M` does not include the `readonly` modifier (§16.4.12), and `E` is one of:
```
It's a little odd to refer to "a readonly struct type" but then talk about the declaration of M "including the readonly modifier", both in the same sentence. Is this because we can't say "readonly method" as it could be a property or indexer? Would "readonly member" work?

In particular, I think that would be slightly helpful in terms of there being multiple places where the modifier can occur for properties - either at the property declaration itself, or the accessor declaration. If we haven't already defined "readonly member" or "readonly function member" perhaps we could do so, then use it here?

I'd be fine to merge as-is and maybe have a clarity follow-up for this though.

BillWagner replied:
Actually, great catch. Property access is covered under 12.8.7.1. Commit coming with a similar change there. 

I'd thought I was being clever by thinking an accessor could be considered a "function member". Better to put the change in 12.8.7.1 and chose more clear language here.

BillWagner committed:
Add non-readonly property accessors.

Make two changes to cover property accessors in structs and the `readonly` modifier.

First, in expressions, add a note that property accessors are classified as function member access: A property access invokes the corresponding function member. So, the same rules apply.
Second, in structs differences, add that accessing a non-readonly setter when that requires creating a temporary is a compile-time error. In the case of a getter, the temporary is created if the getter is not classified as `readonly`.
